File: src/types/video.ts
// Exercise video-related type definitions

export interface ExerciseVideo {
  id: string;
  youtubeId: string;
  youtubeUrl: string;
  title: string;
  description: string;
  category: VideoCategory;
  categoryId: string;
  difficulty: VideoDifficulty;
  duration: number; // in minutes
  equipmentRequired: string[];
  bodyParts: string[];
  tags: string[];
  thumbnailUrl?: string;
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
}

export interface VideoCategory {
  id: string;
  name: string;
  description: string;
  icon?: string;
  createdAt: Date;
  updatedAt: Date;
}

// Form data types for video management
export interface ExerciseVideoFormData {
  title: string;
  description: string;
  youtubeUrl: string;
  categoryId: string;
  difficulty: VideoDifficulty;
  duration: number;
  equipmentRequired: string[];
  bodyParts: string[];
  tags: string[];
}

export interface VideoCategoryFormData {
  name: string;
  description: string;
  icon?: string;
}

// API response types
export interface ExerciseVideoResponse {
  video: ExerciseVideo;
  success: boolean;
  message?: string;
}

export interface ExerciseVideosListResponse {
  videos: ExerciseVideo[];
  total: number;
  page: number;
  limit: number;
}

export interface VideoCategoriesResponse {
  categories: VideoCategory[];
  success: boolean;
}

// Utility types
export type VideoDifficulty = 'beginner' | 'intermediate' | 'advanced';

export interface VideoFilters {
  categoryId?: string;
  difficulty?: VideoDifficulty;
  bodyParts?: string[];
  equipmentRequired?: string[];
  durationMin?: number;
  durationMax?: number;
  searchQuery?: string;
}

export interface VideoSearchParams {
  query?: string;
  category?: string;
  difficulty?: VideoDifficulty;
  bodyPart?: string;
  equipment?: string;
  page?: number;
  limit?: number;
}

// YouTube API types removed - using static thumbnails instead

// Equipment categories
export const EQUIPMENT_TYPES = [
  'None',
  'Resistance Bands',
  'Dumbbells',
  'Exercise Ball',
  'Yoga Mat',
  'Foam Roller',
  'Pilates Ring',
  'Stability Ball',
  'Theraband',
  'Chair',
  'Wall',
] as const;

export type EquipmentType = (typeof EQUIPMENT_TYPES)[number];

// Body parts for exercise targeting
export const EXERCISE_BODY_PARTS = [
  'Full Body',
  'Core',
  'Back',
  'Neck',
  'Shoulders',
  'Arms',
  'Chest',
  'Hips',
  'Legs',
  'Glutes',
  'Calves',
  'Feet',
] as const;

export type ExerciseBodyPart = (typeof EXERCISE_BODY_PARTS)[number];

File: data-manager/internal/models/video.go
package models

import (
	"encoding/json"
	"fmt"
	"time"
)

// VideoCategory represents a video category
type VideoCategory struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Icon        *string   `json:"icon,omitempty"`
	SortOrder   int       `json:"sort_order"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// ExerciseVideo represents an exercise video
type ExerciseVideo struct {
	ID                string    `json:"id"`
	Title             string    `json:"title"`
	Description       string    `json:"description"`
	YoutubeID         string    `json:"youtube_id"`
	YoutubeURL        string    `json:"youtube_url"`
	CategoryID        string    `json:"category_id"`
	Duration          *int      `json:"duration,omitempty"`
	DifficultyLevel   string    `json:"difficulty_level"`
	EquipmentRequired []string  `json:"equipment_required"`
	BodyParts         []string  `json:"body_parts"`
	Tags              []string  `json:"tags"`
	ThumbnailURL      *string   `json:"thumbnail_url,omitempty"`
	CreatedAt         time.Time `json:"created_at"`
	UpdatedAt         time.Time `json:"updated_at"`
	
	// Joined fields
	CategoryName        *string `json:"category_name,omitempty"`
	CategoryDescription *string `json:"category_description,omitempty"`
}

// VideoFormData represents form data for creating/updating videos
type VideoFormData struct {
	Title             string   `json:"title"`
	Description       string   `json:"description"`
	YoutubeURL        string   `json:"youtube_url"`
	CategoryID        string   `json:"category_id"`
	Duration          *int     `json:"duration,omitempty"`
	DifficultyLevel   string   `json:"difficulty_level"`
	EquipmentRequired []string `json:"equipment_required"`
	BodyParts         []string `json:"body_parts"`
	Tags              []string `json:"tags"`
}

// CategoryFormData represents form data for creating/updating categories
type CategoryFormData struct {
	Name        string  `json:"name"`
	Description string  `json:"description"`
	Icon        *string `json:"icon,omitempty"`
	SortOrder   int     `json:"sort_order"`
}

// ToJSON converts the video to JSON string
func (v *ExerciseVideo) ToJSON() (string, error) {
	data, err := json.MarshalIndent(v, "", "  ")
	if err != nil {
		return "", err
	}
	return string(data), nil
}

// ToJSON converts the category to JSON string
func (c *VideoCategory) ToJSON() (string, error) {
	data, err := json.MarshalIndent(c, "", "  ")
	if err != nil {
		return "", err
	}
	return string(data), nil
}

// Validate validates the video form data
func (v *VideoFormData) Validate() error {
	if v.Title == "" {
		return fmt.Errorf("title is required")
	}
	if v.YoutubeURL == "" {
		return fmt.Errorf("youtube URL is required")
	}
	if v.CategoryID == "" {
		return fmt.Errorf("category ID is required")
	}
	if v.DifficultyLevel != "" && v.DifficultyLevel != "beginner" && v.DifficultyLevel != "intermediate" && v.DifficultyLevel != "advanced" {
		return fmt.Errorf("difficulty level must be 'beginner', 'intermediate', or 'advanced'")
	}
	return nil
}

// Validate validates the category form data
func (c *CategoryFormData) Validate() error {
	if c.Name == "" {
		return fmt.Errorf("name is required")
	}
	return nil
}